### PR TITLE
test: Make some integration tests related to raftstore less likely to fail

### LINKED ISSUE #4594: test_region_change_observer fail
## Bug Report

<https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tikv_ghpr_test/detail/tikv_ghpr_test/3752/pipeline/135/>

**TiKV Version:** https://github.com/5kbpers/tikv/tree/50103159042c47962d21d3315a1ac832accf0d39

PTAL @MyonKeminta @hicqu

### LINKED ISSUE #7570: raftstore::test_region_change_observer::test_region_change_observer failed
raftstore::test_region_change_observer::test_region_change_observer

Latest failed builds: 
https://internal.pingcap.net/idc-jenkins/job/tikv_ghpr_test/20938/consoleFull

### PR DESCRIPTION
**Pull request by MyonKeminta**
Signed-off-by: MyonKeminta <MyonKeminta@users.noreply.github.com>

### What problem does this PR solve?

Issue Number: close #7570 , close #4594

Problem Summary:  raftstore::test_region_change_observer::test_region_change_observer is unstable.

### What is changed and how it works?

There's a unstable test in #7570, and according to the logs it was caused by applying snapshot takes too long and the `must_add_peer` runs timeout. This PR tries to fix it by enlarging wait timeout, and hopefully some other tests can also be more stable.

### Related changes

- Need to cherry-pick to the release branch
  - It may be picked to release-4.0 to make the release branch more stable.

### Check List 

Tests 

- Integration test

### Release note 

* No release note.

### CODE REVIEW

File: components/test_raftstore/src/pd.rs [hunk 1]
```diff
@@ -803,8 +803,8 @@ impl TestPdClient {
     }
 
     pub fn must_have_peer(&self, region_id: u64, peer: metapb::Peer) {
-        for _ in 1..500 {
-            sleep_ms(10);
+        for retry in 1..500 {
+            sleep_ms(10 * retry);
```
Comment (youjiali1995) [inline components/test_raftstore/src/pd.rs:807]: The total timeout is 20mins.😂

Comment (MyonKeminta) [reply components/test_raftstore/src/pd.rs:807]: I think I can reduce the retry times to 100 or so, and it won't exceed 1min

File: components/test_raftstore/src/pd.rs [hunk 2]
```diff
@@ -1215,7 +1215,7 @@ impl PdClient for TestPdClient {
             rx.map(|resp| vec![resp])
                 .select(
                     stream::unfold(timer, |timer| {
-                        let interval = timer.delay(Instant::now() + Duration::from_millis(500));
+                        let interval = timer.delay(Instant::now() + Duration::from_millis(5000));
```
Comment (MyonKeminta) [inline components/test_raftstore/src/pd.rs]: Oh, sorry, I accidentally committed this

File: components/test_raftstore/src/pd.rs (hunk 1 above)
Comment (BusyJay) [reply components/test_raftstore/src/pd.rs:807]: Is it the problem of retry time? I think it's scary to run a test case for more than 5 seconds already.

Comment (MyonKeminta) [reply components/test_raftstore/src/pd.rs:807]: @BusyJay According to the log of the latest report in #7570, it finishes applying snapshot just a little more than 5 seconds after starting adding the peer.

**Comment by MyonKeminta** [comment]
@youjiali1995 @BusyJay  PTAL thanks

**Pull request closed without merging**